src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/EnclaveDelegate.Crypto.cs

```diff
@@ -116,7 +114,7 @@ private SqlColumnEncryptionEnclaveProvider GetEnclaveProvider(SqlConnectionAttes
 
             if (sqlColumnEncryptionEnclaveProvider == null)
             {
-                throw SQL.EnclaveProviderNotFound(enclaveType, ConvertAttestationProtocolToString(attestationProtocol));
+                throw SQL.EnclaveProviderNotFound(enclaveType, attestationProtocol.ToString());
```

Comment:
This exception is basically never thrown. If you look at the code above it in the same method, you'll see why. Since the attestation protocol provided in the connection string must pass validation, the default switch case should never be hit, and `sqlColumnEncryptionEnclaveProvider` should never be null. There is no valid use-case that can test this exception.

Reply:
Very odd, I couldn't find any test for that. Seems like a very basic thing to miss. I added a test for invalid protocols.